compiler/rustc_resolve/src/macros.rs

```diff
@@ -695,6 +698,35 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         res.map(|res| (self.get_macro(res).map(|macro_data| macro_data.ext.clone()), res))
     }
 
+    fn report_invalid_crate_level_attr(&mut self, attrs: &[Attribute], name: Symbol) -> bool {
+        for attr in attrs.iter().filter(|attr| attr.style == AttrStyle::Inner) {
+            if attr.has_name(name) {
+                let tcx = self.tcx;
+                tcx.dcx().emit_err(errors::InvalidAttrAtCrateLevel {
```

Comment:
Can the common logic here and in `compiler/rustc_passes/src/check_attr.rs` be shared?

Reply:
seems not easy to share the common logic, but I moved InvalidAttrAtCrateLevel into `rustc_attr` crate, so we only have the duplicated diagnostic resources in different components.